R/migrations.R

```diff
@@ -0,0 +1,93 @@
+run_migration <- function(queue, output_dir, from_version, to_version,
+                          dry_run = TRUE) {
+  output_dir <- normalizePath(output_dir, mustWork = TRUE)
+  tasks <- queue$queue$task_list()
+  status <- queue$queue$task_status(tasks)
+  completed_tasks <- tasks[status == "COMPLETE"]
+  migrations <- lapply(completed_tasks, migrate_task, queue,
+                       from_version, to_version, dry_run)
+  summary <- lapply(migrations, function(migration) {
+    list(
+      id = migration$id,
+      action = migration$action
+    )
+  })
+  summary <- do.call(rbind, summary)
+  summary_path <- file.path(output_dir, "summary.csv")
+  message(sprintf("Saving summary csv %s", summary_path))
+  output_path <- file.path(output_dir, "output.qs")
+  utils::write.csv(summary, summary_path, row.names = FALSE)
+  message(sprintf("Saving output qs %s", output_path))
+  qs::qsave(migrations, output_path, preset = "fast")
+  migrations
+}
+
+migrate_task <- function(task_id, queue, from_version, to_version, dry_run) {
+  message(sprintf("Migrating %s", task_id))
+  res <- queue$queue$task_result(task_id)
+  if (!naomi:::is_hintr_output(res) ||
+      !all(c("plot_data_path", "model_output_path") %in% names(res))) {
+    message(sprintf("Not migrating %s, invalid output format", task_id))
+    return(list(
+      id = task_id,
+      prev_res = res,
+      action = "No change - not migrateable"
+    ))
+  }
+  if (!is.null(res$version) &&
```

Comment:
Why would res$version be null? Do you also need to deal with version is null case as not migrateable?

Reply:
It can be NULL for some very old model fits also. The structure of these outputs has gone through a few iterations. Yep I should deal with that case as not migrateable - will add in a condition for that.

Reply:
Actually had a look at this. We can deal with it if it doesn't have a version number. So as long as there is a `plot_data_path` we can migrate it.